src/adapter/src/coord/ddl.rs

```diff
@@ -756,15 +789,21 @@ impl Coordinator {
         for sink in sinks {
             // Filter out sinks that are currently being dropped. When dropping a sink
             // we send a request to compute to drop it, but don't actually remove it from
-            // `active_subscribes` until compute responds, hence the `dropping` flag.
+            // `active_subscribes` or `active_copy_tos` until compute responds, hence the
+            // `dropping` flag.
             //
             // Note: Ideally we'd use .filter(...) on the iterator, but that would
             // require simultaneously getting an immutable and mutable borrow of self.
             let need_to_drop = self
                 .active_subscribes
                 .get(&sink.global_id)
                 .map(|meta| !meta.dropping)
-                .unwrap_or(false);
+                .unwrap_or(false)
+                || self
+                    .active_copy_tos
+                    .get(&sink.global_id)
+                    .map(|meta| !meta.dropping)
+                    .unwrap_or(false);
```

Comment:
This is a little hard to read like this. Would recommend factoring out into separate variables: `let need_to_drop_subscribe = ...; let need_to_drop_copy_to = ...; if !need_to_drop_subscribe || !need_to_drop_copy_to { ... }`.

Reply:
(Perhaps this disappears if you go with Parker’s suggestion of an ActiveEphemeralSink enum!)

Reply:
updated!